lib/Backend/arm64/ARM64UnwindEncoder.h

```diff
@@ -0,0 +1,157 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#pragma once
+
+class Arm64XdataGenerator
+{
+public:
+    Arm64XdataGenerator();
+
+    void Generate(PULONG prologStart, PULONG prologEnd, PULONG epilogStart, PULONG epilogEnd, PULONG functionEnd, ULONG exceptionHandlerRva = 0, ULONG exceptionData = 0);
+
+    const VOID *GetXdata() const { return &this->m_xdata[0]; }
+    ULONG GetXdataBytes() const { return this->m_xdataBytes; }
+
+    //   4: 2xMOV + BL + SUB for allocation
+    //   4: STP for homing parameters
+    //   6: STP for saving integer non-vols
```

Comment:
>6: STP for saving integer non-vols [](start = 9, length = 34)

On ARM32 we allocate dedicated arguments slot on stack after r11/fp and another one for StackNestedFuncList, I need to do the same thing for ARM64. That means I need to make this 8?

Reply:
Unsure what you're asking here. The calculation you are referencing attempts to determine the maximum number of instructions that we expect to see in a prolog/epilog. I can make that clearer, since I wonder if you were thinking I was counting stack slots not instructions.

Reply:
I've rewritten the comment. Check to see if it makes more sense now.